### DISCUSSION TITLE
Beamforming codebooks and steering vector generation

### DISCUSSION THREAD

**Discussion by NaveenMP** [NVlabs/sionna#370] [General]
Does Sionna support simulated beamforming as part of channel models? More specifically, is there an API which can generate beam steering vector/weights for a given array geometry, with either 3GPP codebook indices or LCS steering angles as input? 

If this feature is not currently supported, it is planned for a future release?

**Comment by jhoydis** [comment]
Hi @NaveenMP,

In short, no. However, it should not be difficult to implement this yourself. You would then only need to project the channel coefficients on the steering vectors to obtain the equivalent channels.
